library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/api/ModResourcePack.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.api;
+
+import net.fabricmc.loader.api.metadata.ModMetadata;
+
+import net.minecraft.resource.ResourcePack;
+
+/**
+ * Interface implemented by mod-provided resource packs.
+ */
+public interface ModResourcePack extends ResourcePack {
```

Comment:
I think we should avoid this scenario where people have to upcast the game types to an high level type in our API. Probably better to just have a method to lookup this additional metadata.

Reply:
I'm not sure to get what you're saying?

Reply:
I think what he's saying is, instead of:
```java
Value value;
if (resPack instanceof ModResourcePack modResPack)
    value = modResPack.value();
else
    throw new Exception("rip");
```
we should have:
```java
Value value = ModResourcePack.getValue(resPack);
```
or... something like that?

Reply:
Well, tbh the first question we should ask is "Is the mod metadata really that useful in resource packs?"

All resource packs are named by default, which makes them already identifiable, and the only use of ModMetadata in the whole resource loader module is:
 - getting the mod path out of it, but that's usually internal, it happens once when registering a built-in resource pack, and once when every mod resource packs are built.
 - generating a default `pack.mcmeta` for mod resource packs
 
 Externally there's literally no use, so is it even that useful to expose this in API?
 The pack.mcmeta generation could be exposed in API, and if a mod needs it it just can fetch the corresponding mod metadata if needed because the only use case of it would be if the mod makes its own resource pack, so no identification issue there.

Reply:
I removed the ModResourcePack interface entirely.